### ISSUE TITLE
warn about pushing "/xy" to a path and overwriting it.

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3923]
````rust
let mut x = PathBuf::from("/foo");
x.push("/bar");
println!("x: {}", x.display()); // "/bar"
````
I just ran into a situation where I did something like this, and it resulted in the path being
`/bar` instead of wanted `/foo/bar`.
What I actually wanted was 
````rust
x.push("bar");
````
so that the path becomes `/foo/bar`

Could we warn about pushing an element to a pathbuf that has the file system root in it, or something like this

**Comment by flip1995** [comment]
This Lint can be implemented in 3 steps:
1. Match on `MethodCall` `push`
2. Match first (`self`) argument of `push` to be `PathBuf`
3. Check the `&str` that is pushed, whether it contains the file system root
